### PR TITLE
B 19832 int add weight stored to api endpoints and DB

### PR DESCRIPTION
**Pull request by r-mettler**
## Summary
This PR adds a new column into the moving_expenses table called weight_stored and updates the moving-expenses API calls in Internal and GHC to use the weight_stored field. This is the first in a series of stories that will give Customers the ability to add amount of weight stored for a moving expense on their PPM shipment and allow Office users to see the amount on the Review Documents page. 

## Verification Steps for the Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Have the Agility acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Has the branch been pulled in and checked out?
- [ ] Have the BL acceptance criteria been met for this change?
- [ ] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Login as a Customer and create and submit a new PPM shipment
2. Login as a Service Counselor and approve the PPM shipment
3. Login as the Customer and click the Upload PPM Documents button
4. Enter the data on the first page click Save & Continue
5. Add a new moving expense using type of "Storage"
6. Inspect in the Chrome dev tools and look at the Network tab and mto_shipments
7. Expand the shipment and look for moving expenses and expand your moving expense
8. You should see a ppm shipment id, the moving expense id (listed as just "id") and the eTag
9. Copy these values off to use in swagger calls
10. Open [swagger for Internal ](http://milmovelocal:3000/swagger-ui/internal.html#/ppm/updateMovingExpense)
11. Enter your copied values for the updateMovingExpense PATCH call
12. Enter weightStored in the body, something like this:   "weightStored": 3000
13. Update the "amount" to be greater than 1
14. Execute the call and verify that you get a success message and see your weightStored value
15. Optionally also look for your moving expense in the moving_expense database table to verify you have a weightStored value
16. Go back to your Customer page and pull the new eTag value
17. Open the [GHC swagger page](http://officelocal:3000/swagger-ui/ghc.html#/ppm/updateMovingExpense)
18. Enter your values for the PATCH call and a different weightStored amount to the body:   "weightStored": 4000
19. Execute and verify success and check DB table
20. Go to the [swagger for Internal ](http://milmovelocal:3000/swagger-ui/internal.html#/ppm/updateMovingExpense)
21. Scroll to the POST call to create a moving expense
22. Enter your ppm shipment id and execute
23. Verify the create was successful

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/getting-started/development/logging).
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/backend/guides/golang-guide#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#zero-downtime-migrations).
- [ ] Have been communicated to #g-database.
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#secure-migrations).

### CODE REVIEW

File: pkg/handlers/internalapi/internal/payloads/payload_to_model.go [hunk 1]
```diff
@@ -273,6 +273,7 @@ func MovingExpenseModelFromUpdate(movingExpense *internalmessages.UpdateMovingEx
 		Amount:            handlers.FmtInt64PtrToPopPtr(movingExpense.Amount),
 		SITStartDate:      handlers.FmtDatePtrToPopPtr(&movingExpense.SitStartDate),
 		SITEndDate:        handlers.FmtDatePtrToPopPtr(&movingExpense.SitEndDate),
+		WeightStored:      handlers.PoundPtrFromInt64Ptr(movingExpense.WeightStored),
```
Comment (MInthavongsay) [inline pkg/handlers/internalapi/internal/payloads/payload_to_model.go:276]: I'm getting the following error
Screenshot 2024-05-13 at 4 26 16 PM

Comment (r-mettler) [reply pkg/handlers/internalapi/internal/payloads/payload_to_model.go]: Forgot to push a change, fixed now.

**Comment by ajlusk** [comment]
Functions as expected all the way through. When I first ran this I encountered the same issue Mike mentioned above but that is no longer occurring. Waiting on failing tests to be passing to approve.

Unrelated to these changes but something that I noticed and other reviewers might too:
In Swagger, the [GHC PATCH request's](http://officelocal:3000/swagger-ui/ghc.html#/ppm/updateMovingExpense) default body includes an approved status and a reason string but returns null for the reason. That is what is supposed to happen, it resets the reason field when the moving expense is approved (moving_expense_updater.go:149).

**Review by MInthavongsay — APPROVED** [review]
Both internal and GHC api calls worked.

**Review by ajlusk — APPROVED** [review]
See above comment.

Is there a reason you removed the example date values in `swagger-def/internal.yaml`?

**Pull request merged**